agent/utils/ephemeral_ports.go

```diff
@@ -129,6 +163,14 @@ func GetHostPortRange(numberOfPorts int, protocol string) (string, error) {
 	}
 	if n != numberOfPorts {
 		return "", fmt.Errorf("%v contiguous host ports unavailable", numberOfPorts)
+	} else {
+		if resultEndPort == endHostPortRange {
+			// we've reached the end of the underlying ephemeral range, so circle back to the start
+			tracker.SetLastAssignedHostPort(startHostPortRange - 1)
```

Comment:
Discussed offline that `startHostPortRange` is assumed to be constant across calls to this function, so resetting tracker to `startHostPortRange - 1` instead of `0` is fine.